### DISCUSSION TITLE
Questions regarding accelerator physics and Ocelot

### DISCUSSION THREAD

**Discussion by irgallard** [ocelot-collab/ocelot#173] [General]
Hello everyone,

I am  a master student ( Artificial Intelligence) currently working on the  topic of applying DRL techniques to Linear Accelerators (IFMIF-DONES in Spain). During my research, I have come across your simulator and I have been actively exploring its capabilities over the past week. I must say, it is very user-friendly, making it easy to get started. However, as someone not deeply immersed in the field of  Accelerator Physics, I have encountered some challenges in fully understanding it. 

I would be immensely grateful if you could offer any guidance or insights that might facilitate my understanding of the physics involved. Some of the questions I have:

- I have read that Twiss parameters are only applicable in the case of periodic conditions in the accelerator. Does it mean periodicity in the repetition of lattice elements? Are these parameters widely used in linear accelerators?
- How is it possible to compute the Twiss parameters without defining a particle array and with a particle array?
- Can these parameters be used for a DRL controller as a first approximation to the control problem?
- Many of the notebook tutorials use ASTRA files. To what extent do I need ASTRA and ASTRA files for Ocelot? 
- In this notebook (https://nbviewer.org/github/ocelot-collab/ocelot/blob/master/demos/ipython_tutorials/accelerator_optim.ipynb). Can I choose the location of the screen to display the beam image? For example, at "marker 2", how would I do it?  
- In the same notebook, does the "s" variable change to something else? I can see its the length of the lattice, but then in the show_e_beam() it takes negative values and is expressed in mm.
- Can I use this code to simulate deuterons instead of electrons?

Thank you for your time,

Galo

**Comment by st-walker** [comment]
>I have read that Twiss parameters are only applicable in the case of periodic conditions in the accelerator. Does it mean periodicity in the repetition of lattice elements? Are these parameters widely used in linear accelerators?

Yes, they are used a lot in both and don't only apply to circular machines.  I use them daily with EuXFEL.

>How is it possible to compute the Twiss parameters without defining a particle array and with a particle array?


Use Twiss instance and the `ocelot.cpbd.optics.twiss` function.   It's in the tutorials.

>Can these parameters be used for a DRL controller as a first approximation to the control problem?

In a general sense I would hope OCELOT does approximate reality to some degree.


>Many of the notebook tutorials use ASTRA files. To what extent do I need ASTRA and ASTRA files for Ocelot?

In general you don't, but we use them a lot for the EuXFEL because of various physics related reasons that may or may not apply to you.  It's just a widely used code in injectors that we need for the first 3.2m of the EuXFEL from the cathode.  You can load from anything.


>In this notebook (https://nbviewer.org/github/ocelot-collab/ocelot/blob/master/demos/ipython_tutorials/accelerator_optim.ipynb). Can I choose the location of the screen to display the beam image? For example, at "marker 2", how would I do it?

When you track, you track to the last point in the lattice.  If you simply end the lattice where you want the "screen" to be, then you can look at the particle beam at that point.   You can also use physics processes to dump the beam at arbitrary markers in the machine (`ocelot.cpbd.physics_proc.SaveBeam`).

>In the same notebook, does the "s" variable change to something else? I can see its the length of the lattice, but then in the show_e_beam() it takes negative values and is expressed in mm.

This is the local longitudinal offset, i.e. the particle's coordinate with respect to the reference particle (typically something approximating the centre of the beam).  So they are two different things, although it is confusing in this case slightly because they share the same symbol.

>Can I use this code to simulate deuterons instead of electrons?

All particles in OCELOT are assumed to be electrons (always have negative charge and mass of electron is used throughout for relativistic gamma, etc.).  Of course in principle this could be changed, with the appropriate refactoring of the existing codebase, but this is a lot of work.

**Reply by irgallard** [reply]
Thanks for the replies, this helped a lot!
